src/components/VideoPlayer/BaseVideoPlayer.tsx

```diff
@@ -172,22 +178,40 @@ function BaseVideoPlayer({
         });
     }, [currentVideoPlayerRef, handleFullscreenUpdate, handlePlaybackStatusUpdate]);
 
+    useLayoutEffect(() => 
+        () => {
+            if(shouldUseSharedVideoElement || videoPlayerRef.current !== currentVideoPlayerRef.current) {
+                return;
+            }
+            currentVideoPlayerRef.current = null;
+        }
+    , [currentVideoPlayerRef, shouldUseSharedVideoElement]);
+
     useEffect(() => {
         if (!isUploading || !videoPlayerRef.current) {
             return;
         }
 
         // If we are uploading a new video, we want to immediately set the video player ref.
+        if(currentVideoPlayerRef.current) {
+            pauseVideo();
+        }
         currentVideoPlayerRef.current = videoPlayerRef.current;
-    }, [url, currentVideoPlayerRef, isUploading]);
+    }, [url, currentVideoPlayerRef, isUploading, pauseVideo]);
 
     // update shared video elements
     useEffect(() => {
-        if (shouldUseSharedVideoElement || url !== currentlyPlayingURL || isFullScreenRef.current) {
+        if (shouldUseSharedVideoElement || url !== currentlyPlayingURL) {
```

Comment:
@s77rt 
Let's check this test case.
1. play `video1`.
2. play `video2`.
3. press fullscreen mode button of `video1`.
4. exit fullscreen mode with pressing esc key. (don't play video)
5. press play button of `video1`.

Before fix:
 `video2` is played.

This error is occurred because `currentVideoPlayerRef` is not set when enter fullscreen mode.
So I've deleted above condition.
Please let me know your suggestion.